shared/message/input/JoystickEvent.proto

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2013 NUbots <nubots@nubots.net>
+ */
+
+syntax = "proto3";
+
+package message.input;
+
+import "google/protobuf/timestamp.proto";
+
+message JoystickEvent {
+    enum EventType {
+        UNKNOWN = 0;
+        BUTTON  = 1;    // Button press/release
+        AXIS    = 2;    // Joystick movement
+        INIT    = 128;  // Initial state of device
+    }
+
+    enum Axis {
+        LEFT_JOYSTICK_HORIZONTAL  = 0;
+        LEFT_JOYSTICK_VERTICAL    = 1;
+        L2                        = 2;
```

Comment:
Why's L2 an Axis? Should i not be a button?

Reply:
Probably because the driver reports the L2/R2 events as being "axis" events.

Reply:
L2 and R2 are variable, not just on off